internal/resource-request-operator/offerUpdater.go

```diff
@@ -0,0 +1,146 @@
+		return false
+	}
+	err := func(obj interface{}) error {
+		defer u.queue.Done(obj)
+		var clusterID string
+		var ok bool
+		if clusterID, ok = obj.(string); !ok {
+			// As the item in the workqueue is actually invalid, we call
+			// Forget here else we'd go into a loop of attempting to
+			// process a work item that is invalid.
+			u.queue.Forget(obj)
+			return fmt.Errorf("error getting object %v from OfferUpater queue. It is not a string", obj)
+		}
+		// check if the clusterID is still valid and can be processed
+		if !u.isValidClusterID(clusterID) {
+			// if clusterID is not valid means that the peering has been dropped or something went wrong.
+			// so we call Forget and RemoveClusterID in order to invalidate it.
+			u.broadcasterInt.RemoveClusterID(clusterID)
+			return fmt.Errorf("clusterID %s is no more valid and will be deleted", clusterID)
+		}
+		if err := u.updateOffer(clusterID); err != nil {
+			// Put the item back on the workqueue to handle any transient errors.
+			u.queue.AddRateLimited(clusterID)
+			return fmt.Errorf("error during updating ResourceOffer for cluster %s: %w", clusterID, err)
+		}
+		return nil
+	}(obj)
+	if err != nil {
+		klog.Errorf("Error occurred during ResourceOffer update %s", err)
+		return true
+	}
+	klog.Infof("Update cluster %s processed", obj.(string))
+
+	// requeue after timeout seconds
+	u.queue.AddAfter(obj, requeueTimeout+time.Duration(getRandomDelta())*time.Second)
+	return true
+}
+
+func (u *OfferUpdater) updateOffer(clusterID string) error {
+	offerList := &sharingv1alpha1.ResourceOfferList{}
+	if err := u.Client.List(context.Background(), offerList, client.MatchingLabels{
+		discovery.ClusterIDLabel: clusterID,
+	}); err != nil {
+		return err
+	}
+	if len(offerList.Items) > 1 {
+		return fmt.Errorf("too many ResourceOffers for cluster %s", clusterID)
+	} else if len(offerList.Items) == 0 {
+		return fmt.Errorf("no ResourceOffer for cluster %s", clusterID)
+	}
+	offer := &offerList.Items[0]
+	resources := u.broadcasterInt.ReadResources(clusterID)
+	offer.Spec.ResourceQuota.Hard = resources.DeepCopy()
+	if err := u.Client.Update(context.Background(), offer); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Push add new clusterID to update queue which will be processes as soon as possible.
+func (u *OfferUpdater) Push(clusterID string) {
+	u.queue.Add(clusterID)
+}
+
+// Remove remove a specified clusterID from the update queue and it will be no more processed.
+func (u *OfferUpdater) Remove(clusterID string) {
+	u.queue.Forget(clusterID)
+	klog.Infof("Removed cluster %s from update queue", clusterID)
+}
+
+func (u *OfferUpdater) isValidClusterID(clusterID string) bool {
+	resourceRequestList := &discoveryv1alpha1.ResourceRequestList{}
+	err := u.Client.List(context.Background(), resourceRequestList)
+	if err != nil || len(resourceRequestList.Items) != 1 {
+		return false
+	}
+	return true
+}
+
+func getRandomDelta() int64 {
+	max := new(big.Int)
+	max.SetInt64(int64(60))
```

Comment:
Would avoid hardcoding the value here and define an appropriate constant with the associated comment.